src/guiapi/src/ScreenShot.cpp

```diff
@@ -53,62 +56,57 @@ static const unsigned char bmp_header[] = {
 };
 
 static void mirror_buffer(uint8_t *buffer) {
-    // Y-axis mirror image - because BMP pixel format has base origin in left-bottom corner and st7789v in left-upper corner
-
-    // TODO: BMP headers have to know that we are using 2B pixels. Now it is only in [bits per pixel] and clearly it's not enough.
-
-    for (int row = 0; row < ST7789V_BUFF_ROWS / 2; row++) {
-        for (int col = 0; col < ST7789V_COLS * ST7789V_BYTES_PER_PIXEL; col += ST7789V_BYTES_PER_PIXEL) {
-            for (int chan = 0; chan < ST7789V_BYTES_PER_PIXEL; chan++) {
-                std::swap(buffer[buffer[row * ST7789V_COLS * ST7789V_BYTES_PER_PIXEL + col + chan]], buffer[(ST7789V_BUFF_ROWS - row - 1) * ST7789V_COLS * ST7789V_BYTES_PER_PIXEL + col + chan]);
+    // Y-axis mirror image - because BMP pixel format has base origin in left-bottom corner not in left-top like on displays
+    for (int row = 0; row < buffer_rows / 2; row++) {
+        for (int col = 0; col < display::GetW() * bytes_per_pixel; col += bytes_per_pixel) {
+            for (int chan = 0; chan < bytes_per_pixel; chan++) {
+#ifdef USE_ST7789
```

Comment:
I think this should always work since it's byte-wise, irregardless of display. I'd remove the ifdef